target/hexagon/idef-parser/idef-parser.lex

```diff
@@ -2,6 +2,7 @@
 %option 8bit reentrant bison-bridge
 %option warn nodefault
 %option bison-locations
+%option never-interactive
```

Comment:
Why?

Reply:
Flex has the ability to be used interactively by default. This adds a call to `isatty(fileno(...))` to check whether or not it is used interactively. `fileno` is undeclared in C11 and so I added the `%option never-interactive` to get rid of the `isatty(fileno(...))` check.

I realize now that this necessary as QEMU uses `gnuC99` which includes `fileno`. I think its a good idea to remove the `C11` flag then and use `gnuC99` to comply with QEMU coding standards. This will also mean changing the C11 macro `_Generic` for their version `QEMU_GENERIC`.

Reply:
> Flex has the ability to be used interactively by default. This adds a call to `isatty(fileno(...))` to check whether or not it is used interactively. `fileno` is undeclared in C11 and so I added the `%option never-interactive` to get rid of the `isatty(fileno(...))` check.

I see.

> I realize now that this necessary as QEMU uses `gnuC99` which includes `fileno`.

You mean "is *not* necessary"?

> I think its a good idea to remove the `C11` flag then and use `gnuC99` to comply with QEMU coding standards. This will also mean changing the C11 macro `_Generic` for their version `QEMU_GENERIC`.

Ideally we should not add flags such as `-std=c11` and use whatever is globally provided. That's why I asked the current command line in the other command.

:+1: for `QEMU_GENERIC`.

Reply:
> You mean "is not necessary"?
Oh yes, my bad.